### PR TITLE
use conditional mutations to remove hosting request

### PR DESCRIPTION
**Pull request by keith-turner**

### CODE REVIEW

**Comment by keith-turner** [comment]
Still working on other changes for #3858, but did not want to put them all in the same commit.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java [hunk 1]
```diff
@@ -1191,10 +1195,20 @@ public void evaluateOnDemandTabletsForUnload() {
...
+          tabletsMutator.mutateTablet(ke).requireLocation(myLocation).deleteHostingRequested()
+              .submit(tm -> !tm.getHostingRequested());
+        });
+
+        tabletsMutator.process().forEach((extent, result) -> {
+          if (result.getStatus() != Ample.ConditionalResult.Status.ACCEPTED) {
+            var loc = Optional.ofNullable(result.readMetadata()).map(TabletMetadata::getLocation)
+                .orElse(null);
+            log.debug("Failed to clear hosting request marker for {} location in metadata:{}",
+                extent, loc);
```
Comment (dlmarion) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java:1210]:
```suggestion
        if (!loc.equals(myLocaion) {
            log.debug("Failed to clear hosting request marker for {} location in metadata:{}",
                extent, loc);
        }
```
I wonder if we should only print this message if the location is *this* location. I guess in theory it's possible that the tablet was unloaded and re-hosted on another tablet server?

**Review by dlmarion — APPROVED** [review]

File: server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java (hunk 1 above)
Comment (keith-turner) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java:1210]: > I guess in theory it's possible that the tablet was unloaded and re-hosted on another tablet server?

Yeah this runs outside of the tablet code so its possible that the tablet unloads between the check of tablet status and writing the conditional mutation.  So this log message could be noise and the suggested change would probably be good to do later on after testing more extensively.

**Comment by keith-turner** [comment]
While looking into #3858 I found there were two places in tablet server that were removing hosting request.  So I took the code from this PR and put it in a function to call in both places.  Did this in b4ecd1a and pushed that commit directly to elasticity branch.

**Pull request merged**